db/write_callback_test.cc

```diff
@@ -84,7 +84,32 @@ class MockWriteCallback : public WriteCallback {
   bool AllowWriteBatching() override { return allow_batching_; }
 };
 
-TEST_F(WriteCallbackTest, WriteWithCallbackTest) {
+class WriteCallbackPTest
+    : public WriteCallbackTest,
+      public ::testing::WithParamInterface<
+          std::tuple<bool, bool, bool, bool, bool, bool, bool>> {
+ public:
+  void SetUp() override {
+    unordered_write_ = std::get<0>(GetParam());
+    seq_per_batch_ = std::get<1>(GetParam());
+    two_queues_ = std::get<2>(GetParam());
+    allow_parallel_ = std::get<3>(GetParam());
+    allow_batching_ = std::get<4>(GetParam());
+    enable_WAL_ = std::get<5>(GetParam());
+    enable_pipelined_write_ = std::get<6>(GetParam());
```

Comment:
We could do `std::tie(unordered_write_, ...) = GetParam();` to avoid having to call `GetParam` multiple times.